### PR TITLE
complete text, duration, midi, npy, wav and data_dir and left the data.sh to complish

### PR DESCRIPTION
**Pull request by DrWelles**
i may finish these 2days later

### CODE REVIEW

**Comment by ftshijt** [comment]
please add corresponding package requirements at `setup.py` (e.g., miditoolkit)

File: egs/kiritan/local/prep_segments.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            utt_id = recording_id
+            sys.stdout.write(
+                "{} {} {} {}\n".format(utt_id, seg_id, segment_begin, segment_end)
+            )
+            seg_id += 1
+
+
+
+
+        # # labels = hts.load(lab_path)
```
Comment (ftshijt) [inline egs/kiritan/local/prep_segments.py:53]: Please remove these lines with comments

File: egs/kiritan/local/prep_segments.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    args = get_parser().parse_args(sys.argv[1:])
+    Window_size = 60 * 1e-3
+    Window_shift = 30 * 1e-3
+
+    labels = []
+    with open(args.wav_scp) as f:
+        for line in f:
+            recording_id, path = line.split()
+            lab_path = path.replace("wav/", "mono_label/").replace(".wav", ".lab")
+            # lab_path = path.replace("wav/", "lab/").replace(".wav", ".lab")
```
Comment (ftshijt) [inline egs/kiritan/local/prep_segments.py:28]: remove debug information

File: egs/kiritan/local/prep_segments.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        description="Prepare segments from HTS-style alignment files",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("wav_scp", type=str, help="wav scp file")
+    return parser
+
+
+if __name__ == "__main__":
+    args = get_parser().parse_args(sys.argv[1:])
+    Window_size = 60 * 1e-3
```
Comment (ftshijt) [inline egs/kiritan/local/prep_segments.py:20]: This should be aligned with some arguments, the window size should not be fixed here (so as the window shift next line)

File: egs/kiritan/local/prep_segments.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    Window_shift = 30 * 1e-3
+
+    labels = []
+    with open(args.wav_scp) as f:
+        for line in f:
+            recording_id, path = line.split()
```
Comment (ftshijt) [inline egs/kiritan/local/prep_segments.py:26]: I don't think this is a robust function though. What if we have

utt1 "place_with_ space _/wav"

File: egs/kiritan/local/prep_segments.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                for line in lab_f:
+                    start, end, phone = line.split(' ')
+                    # print(line)
+                    labels.append([ float(start), float(end), phone])
+
+        segments = []
+        max_time = labels[-1][1]
+        seg_id = 0
+        while seg_id*Window_shift + Window_size <= max_time:
+            i = seg_id * Window_shift
```
Comment (ftshijt) [inline egs/kiritan/local/prep_segments.py:41]: window size and window shift, only 60ms for a segment?

File: muskit/fileio/midi_scp.py [hunk 6]
```diff
@@ -5,48 +5,56 @@
...
-
-        return rate, array
+        # return miditoolkit.midi.parser.MidiFile(self.data[key])
+        midi_obj = miditoolkit.midi.parser.MidiFile(self.data[key])
+        seq = []
+        if self.rep == "representation":
+            seq = midi_to_noteseq(midi_obj, self.dtype, self.rate)
+        return seq
+        # wav = self.data[key]
+        # if self.normalize:
```
Comment (ftshijt) [inline muskit/fileio/midi_scp.py:49]: remove comments

File: muskit/fileio/utils.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    midi.instruments.append(inst)
+    # write tempo
+    tempo_changes = []
+    for st, bpm in tempos:
+        tempo_changes.append(miditoolkit.midi.containers.TempoChange(bpm, st))
+    midi.tempo_changes = tempo_changes
+    return midi
+
+
+if __name__ == '__main__':
```
Comment (ftshijt) [inline muskit/fileio/utils.py:81]: move this part to test functions later (we do not have them yet)

**Review by ftshijt — CHANGES_REQUESTED** [review]

**Comment by ftshijt** [comment]
Please reflect on the comments and fix the conflict. Then I will merge it.

**Comment by DrWelles** [comment]
well, it seems that it's quiet slow and may be faster....

File: setup.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< Updated upstream
```
Comment (ftshijt) [inline setup.py:1]: There seems some conflicts not fixed

File: muskit/fileio/midi_scp.py [hunk 9]
```diff
@@ -124,3 +118,126 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.fscp.close()
+>>>>>>> Stashed changes
```
Comment (ftshijt) [inline muskit/fileio/midi_scp.py:243]: same for here

File: muskit/fileio/read_text.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< Updated upstream
```
Comment (ftshijt) [inline muskit/fileio/read_text.py:1]: Please check `<<<<<` to make sure no errors

**Review by ftshijt — CHANGES_REQUESTED** [review]

**Comment by ftshijt** [comment]
> well, it seems that it's quiet slow and may be faster....

Which part is slow? Format wav or preparing at the beginning?

**Comment by DrWelles** [comment]
> > well, it seems that it's quiet slow and may be faster....
> 
> Which part is slow? Format wav or preparing at the beginning?

format wav&midi, the text and duration is fast enough

File: muskit/fileio/midi_scp.py [hunk 11]
```diff
@@ -1,52 +1,51 @@
+<<<<<<< Updated upstream
```
Comment (ftshijt) [inline muskit/fileio/midi_scp.py:1]: Because of the merge conflict, there remains such FIXME code, please fix them

**Review by ftshijt — CHANGES_REQUESTED** [review]

**Comment by ftshijt** [comment]
There are still some setting issues, I will follow this PR, so will not merge until I fix the issues

**Pull request merged**